test/hulaaki/client_tcp_test.exs

```diff
@@ -382,4 +382,27 @@ defmodule Hulaaki.ClientTCPTest do
     post_disconnect pid
   end
 
+  test "on_subscribed_publish_ack with qos 2, should not receive subscribed_publish_ack message", %{client_pid: pid} do
+    pre_connect pid
+
+    options = [topics: ["awesome"], qoses: [1]]
+    SampleClient.subscribe(pid, options)
+
+    spawn fn ->
+      {:ok, pid2} = SampleClient.start_link(%{parent: self()})
+
+      options = [client_id: "another-name-8234", host: TestConfig.mqtt_host, port: TestConfig.mqtt_port]
+      SampleClient.connect(pid2, options)
+
+      options = [topic: "awesome", message: "a message",
+                 dup: 0, qos: 1, retain: 2]
```

Comment:
I just realised that this was a bad test (it's def. not setting qos to 2 here). My bad that I let this through, will get rid of this in a future commit.